logstash-core/src/main/java/org/logstash/config/ir/CompiledPipeline.java

```diff
@@ -136,14 +140,26 @@ public CompiledPipeline(
     }
 
     /**
-     * This method contains the actual compilation of the {@link Dataset} representing the
-     * underlying pipeline from the Queue to the outputs.
-     * @return Compiled {@link Dataset} representation of the underlying {@link PipelineIR} topology
+     * Perform the actual compilation of the {@link Dataset} representing the
+     * underlying pipeline from the Queue to the outputs using the
+     * unordered  execution model.
+     * @return CompiledPipeline.CompiledExecution the compiled pipeline
      */
-    public Dataset buildExecution() {
-        return new CompiledPipeline.CompiledExecution().toDataset();
+    public CompiledPipeline.CompiledExecution buildExecution() {
+        return buildExecution(false);
     }
 
+    /**
+     * Perform the actual compilation of the {@link Dataset} representing the
+     * underlying pipeline from the Queue to the outputs using the ordered or
+     * unordered  execution model.
+     * @return CompiledPipeline.CompiledExecution the compiled pipeline
```

Comment:
build fails without doc for the param

```suggestion
     * @param orderedExecution determines whether to build an execution that enforces order or not
     * @return CompiledPipeline.CompiledExecution the compiled pipeline
```

Reply:
ah, yes.